src/main/java/ch/bfh/bti7081/s2018/green/presenters/MedicationPrescriptionPresenter.java

```diff
@@ -0,0 +1,129 @@
+package ch.bfh.bti7081.s2018.green.presenters;
+
+import ch.bfh.bti7081.s2018.green.models.entities.Medication;
+import ch.bfh.bti7081.s2018.green.views.MedciationPrescriptionView;
+import ch.bfh.bti7081.s2018.green.views.enumerations.TxtBtnSaveEdit;
+import com.vaadin.ui.*;
+
+import javax.persistence.*;
+import java.time.LocalDateTime;
+
+public class MedicationPrescriptionPresenter extends MedciationPrescriptionView {
```

Comment:
Why extend the view?

Reply:
Vaadin didn't want to allow me of getting recent values of changed field somehow.
So this was a way to fix it.

Furthermore:
It shares some method with the View (add/remove components) that I didn't want to rewrite.
If would rewrite them in the presenter it self, the presenter must implement CustomLayout. This would be also against the sense of extract UI logic from the view.
I know inheritance does technically also call a view instance (whicht does inherit from CustomLayout) when creating the presenter - so not really different. But good for code lazyness.

There are models from active and passive MVP pattern that would maybe adress similar cases when a presenter with logic needs to access, remove and add fields to a view and how to implement this. But I dind't fully understand the differences.

Maybe we need to discuss this later. Low prio in my opinion.